esp-hal/src/uart.rs

```diff
@@ -452,19 +436,73 @@ where
     }
 }
 
+impl<'d, T> UartTx<'d, T, Blocking>
+where
+    T: Instance + 'd,
+{
+    /// Create a new UART TX instance in [`Blocking`] mode.
+    pub fn new<TX: OutputPin>(
+        uart: impl Peripheral<P = T> + 'd,
+        clocks: &Clocks,
+        interrupt: Option<InterruptHandler>,
+        tx: impl Peripheral<P = TX> + 'd,
+    ) -> Self {
+        Self::new_with_config(uart, Default::default(), clocks, interrupt, tx)
+    }
+
+    /// Create a new UART TX instance with configuration options in
+    /// [`Blocking`] mode.
+    pub fn new_with_config<TX: OutputPin>(
+        uart: impl Peripheral<P = T> + 'd,
+        config: Config,
+        clocks: &Clocks,
+        interrupt: Option<InterruptHandler>,
+        tx: impl Peripheral<P = TX> + 'd,
+    ) -> Self {
+        crate::into_ref!(tx);
+        tx.set_to_push_pull_output(Internal);
+        tx.connect_peripheral_to_output(T::tx_signal(), Internal);
+
+        let (uart_tx, _) =
+            Uart::<'d, T, Blocking>::new_with_config_inner(uart, config, clocks, interrupt).split();
+
+        uart_tx
+    }
+}
+
 impl<'d, T, M> UartRx<'d, T, M>
 where
     T: Instance,
     M: Mode,
 {
-    fn new_inner() -> Self {
+    #[cfg(not(esp32))]
+    fn new_inner(symbol_len: u8) -> Self {
```

Comment:
iirc it's possible to cfg parameters of a function, so I don't think we need to duplicate this.